src/array/cpu/spmm.h

```diff
@@ -315,7 +328,10 @@ template <typename DType> constexpr bool CopyRhs<DType>::use_rhs;
 //////////////////////////////// Reduce operators on CPU ////////////////////////////////
 template <typename DType>
 struct Max {
-  static constexpr DType zero = std::numeric_limits<DType>::lowest();
+  static constexpr DType zero = (
+      std::numeric_limits<DType>::has_infinity ?
+      -std::numeric_limits<DType>::infinity() :
+      std::numeric_limits<DType>::lowest());
```

Comment:
Why don't we just use `lowest` and `max`?

Reply:
Now I'm only using infinity.